SplunkRumWorkspace/SplunkRum/SplunkRum/SessionBasedSampler.swift

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import OpenTelemetrySdk
+import OpenTelemetryApi
+
+class SessionBasedSampler {
+    static var probability: Double = 1.0
+    static var sessionCount: Int = 0
```

Comment:
If I'm understanding the usage of this field correctly we don't really need to count the sessions in order to sample them; see later comments.

Reply:
True if we use a random number then we can remove this entirely along with the setup code for this.